tests/lib/rules/require-unicode-regexp.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * @fileoverview Tests for require-unicode-regexp rule.
+ * @author Toru Nagashima
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require("../../../lib/rules/require-unicode-regexp");
+const RuleTester = require("../../../lib/testers/rule-tester");
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+    parserOptions: { ecmaVersion: 2015 }
+});
+
+ruleTester.run("require-unicode-regexp", rule, {
+    valid: [
+        "/foo/u",
+        "/foo/gimuy",
+        "RegExp('', 'u')",
+        "new RegExp('', 'u')",
+        "RegExp('', 'gimuy')",
+        "new RegExp('', 'gimuy')",
+        "const flags = 'u'; new RegExp('', flags)",
+        "const flags = 'g'; new RegExp('', flags + 'u')",
+        "const flags = 'gimu'; new RegExp('foo', flags.slice(1))",
+        "new RegExp('', flags)",
+        "function f(flags) { return new RegExp('', flags) }",
+        "function f(RegExp) { return new RegExp('foo') }"
+    ],
+    invalid: [
+        {
+            code: "/foo/",
+            errors: [{ messageId: "requireUFlag" }]
+        },
+        {
+            code: "/foo/gimy",
+            errors: [{ messageId: "requireUFlag" }]
+        },
+        {
+            code: "RegExp('foo')",
+            errors: [{ messageId: "requireUFlag" }]
+        },
+        {
+            code: "RegExp('foo', '')",
+            errors: [{ messageId: "requireUFlag" }]
+        },
+        {
+            code: "RegExp('foo', 'gimy')",
+            errors: [{ messageId: "requireUFlag" }]
+        },
+        {
+            code: "new RegExp('foo')",
+            errors: [{ messageId: "requireUFlag" }]
+        },
+        {
+            code: "new RegExp('foo', '')",
+            errors: [{ messageId: "requireUFlag" }]
+        },
+        {
+            code: "new RegExp('foo', 'gimy')",
+            errors: [{ messageId: "requireUFlag" }]
+        },
+        {
+            code: "const flags = 'gi'; new RegExp('foo', flags)",
+            errors: [{ messageId: "requireUFlag" }]
+        },
+        {
+            code: "const flags = 'gimu'; new RegExp('foo', flags.slice(0, -1))",
```

Comment:
I'm confused. The documentation says this test should be valid (i.e., rule ignores these cases where flags is non-constant variable), but this test is in the invalid section. Seems either this or the documentation needs to change.

Reply:
My word choice seemed bad. That ignores a `RegExp` call if it couldn't evaluate the 2nd argument to a static value in the file. In the test case, `flags.slice(0, -1)` is evaluated to `'gim'` under the assumption that the standard APIs are valid.

Reply:
Initially I assumed this wouldn't be determined statically, but I spent a few minutes looking at your `eslint-utils` module, specifically the `getStaticValue` implementation, and I am thoroughly impressed with how powerful it is :clap: